pynetdicom/transport.py

```diff
@@ -146,20 +149,24 @@ def connect(self, address):
                     server_side=False,
                     server_hostname=server_hostname,
                 )
+            # Set ae connection timeout
+            self.socket.settimeout(self.assoc.connection_timeout)
             # Try and connect to remote at (address, port)
             #   raises socket.error if connection refused
             self.socket.connect(address)
+            # Clear ae connection timeout
+            self.socket.settimeout(None)
             # Trigger event - connection open
             evt.trigger(self.assoc, evt.EVT_CONN_OPEN, {'address' : address})
             self._is_connected = True
             # Evt2: Transport connection confirmation
             self.event_queue.put('Evt2')
-        except (socket.error, socket.timeout) as exc:
+        except OSError as exc:
             # Log connection failure
             LOGGER.error(
                 "Association request failed: unable to connect to remote"
             )
-            LOGGER.error("TCP Initialisation Error: Connection refused")
+            LOGGER.error(f"TCP Initialisation Error: {exc}")
```

Comment:
ECONNREFUSED is one of a few cases that could be caught here - I thought it would be nice to be able to distinguish these cases in the log